src/monexamine.cpp

```diff
@@ -380,11 +395,12 @@ void monexamine::attach_or_remove_saddle( monster &z )
 {
     if( z.has_effect( effect_saddled ) ) {
         z.remove_effect( effect_saddled );
-        item riding_saddle( "riding_saddle", 0 );
-        g->u.i_add( riding_saddle );
+        g->u.i_add( z.tack_item.value() );
+        z.tack_item = cata::nullopt;
     } else {
         z.add_effect( effect_saddled, 1_turns, num_bp, true );
-        g->u.use_amount( "riding_saddle", 1 );
+        z.tack_item = g->u.i_at( tack_pos() );
+        g->u.use_amount( z.tack_item.value().typeId(), 1 );
```

Comment:
This may consume a different item (of the same type). But if the user selects e.g. a damaged item, this may consume an undamaged item they also have in their inventory.

Reply:
I don't see anything like that happening in testing with different tack items in my inventory with different damage states. How could the item position give it a different item?

Reply:
The item position is the index of the item stack within the inventory. If your character has several stacks of saddles (damage, inscribed, ...), they each have a different index value. But they all have the very same item type id (`typeId()` == "saddle").

You remove the saddle from the inventory via the id (`g->u.use_amount( "saddle", 1 )`) - how can the game possible know which of the many saddles in the inventory it should remove?

----

Which of the item stacks did you select in your tests? I image `g->u.use_amount` removes the first suitable item it finds in the inventory, so it should always remove the item from the first stack.

Reply:
This should be solved by refactoring it to use `item_location`.